File: netlify/functions/lib/scrapers/eztv.js
/**
 * EZTV scraper - HTML scraping, primarily for TV shows.
 */
const client = require('../client');
const { parseSeeders, parseSize, cleanName } = require('../utils');

const NAME = 'EZTV';
const BASE = 'https://eztvx.to';

async function search(query) {
  const encoded = query.replace(/\s+/g, '-');
  const url = `${BASE}/search/${encodeURIComponent(encoded)}`;
  const $ = await client.getSoup(url);
  if (!$) return [];

  const results = [];
  const table = $('table.forum_header_border');
  if (!table.length) return [];

  table.find('tr.forum_header_border').each((_, row) => {
    const cols = $(row).find('td');
    if (cols.length < 5) return;

    const nameTag = $(cols[1]).find('a.epinfo').first();
    const name = cleanName(nameTag.text());
    if (!name) return;

    const magnetTag = $(cols[2]).find('a.magnet, a[href^="magnet:"]').first();
    if (!magnetTag.length) return;
    const magnet = magnetTag.attr('href') || '';
    if (!magnet.startsWith('magnet:')) return;

    const sizeText = cols.length > 3 ? $(cols[3]).text().trim() : '';
    const seeders = cols.length > 5 ? parseSeeders($(cols[5]).text()) : 0;

    results.push({
      name,
      size: sizeText || 'N/A',
      size_gb: parseSize(sizeText),
      seeders,
      magnet,
      provider: NAME,
    });
  });
  return results;
}

module.exports = { name: NAME, base_url: BASE, search };

File: scrapers/eztv.py
"""EZTV scraper - HTML scraping, primarily for TV shows."""

import re
from urllib.parse import quote
from scrapers.base import BaseScraper
from scrapers import client, utils


class Source(BaseScraper):
    name = 'EZTV'
    base_url = 'https://eztvx.to'

    def search(self, query):
        search_query = query.replace(' ', '-')
        url = f'{self.base_url}/search/{quote(search_query)}'
        soup = client.get_soup(url)
        if not soup:
            return []
        results = []
        table = soup.select_one('table.forum_header_border')
        if not table:
            return []
        for row in table.select('tr.forum_header_border'):
            cols = row.select('td')
            if len(cols) < 5:
                continue
            # Name from second column link
            name_tag = cols[1].select_one('a.epinfo')
            name = utils.clean_name(name_tag.get_text()) if name_tag else ''
            if not name:
                continue
            # Magnet link from third column
            magnet_tag = cols[2].select_one('a.magnet, a[href^="magnet:"]')
            if not magnet_tag:
                continue
            magnet = magnet_tag.get('href', '')
            if not magnet.startswith('magnet:'):
                continue
            # Size from fourth column
            size_text = cols[3].get_text(strip=True) if len(cols) > 3 else ''
            # Seeders from fifth column
            seeders = utils.parse_seeders(cols[5].get_text()) if len(cols) > 5 else 0
            results.append({
                'name': name,
                'size': size_text or 'N/A',
                'size_gb': utils.parse_size(size_text),
                'seeders': seeders,
                'magnet': magnet,
                'provider': self.name,
            })
        return results
